toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -499,24 +499,24 @@ func calculateTotalPackages(packages []string, installRoot string) (totalPackage
 	return
 }
 
+// addMachineID creates the /etc/machine-id file in the installChroot
 func addMachineID(installChroot *safechroot.Chroot) (err error) {
+	// From https://www.freedesktop.org/software/systemd/man/machine-id.html:
+	// For operating system images which are created once and used on multiple
+	// machines, for example for containers or in the cloud, /etc/machine-id
+	// should be an empty file in the generic file system image. An ID will be
+	// generated during boot and saved to this file if possible.
+
 	const (
-		squashErrors = false
-		setupProgram = "/bin/systemd-machine-id-setup"
+		squashErrors  = false
+		setupProgram  = "touch"
+		machineIDFile = "/etc/machine-id"
 	)
 
-	// Check if systemd-machine-id-setup is present before invoking it,
-	// some images will not use systemd (such as a container)
-	exists, _ := file.PathExists(filepath.Join(installChroot.RootDir(), setupProgram))
-	if !exists {
-		logger.Log.Debugf("'%s' not found inside chroot '%s', skipping adding machine ID", setupProgram, installChroot.RootDir())
-		return
-	}
-
 	ReportAction("Configuring machine id")
 
 	err = installChroot.UnsafeRun(func() error {
-		return shell.ExecuteLive(squashErrors, setupProgram)
+		return shell.ExecuteLive(squashErrors, setupProgram, machineIDFile)
```

Comment:
Do we need to apply any specific permissions to this file?

Reply:
Probably should. Default permissions are `644` root ownership, but this should be explicitly set.

Reply:
Using `install` now with explicit file permissions instead of `touch`